### PR TITLE
PR: Implement Kite installation logic

### PR DESCRIPTION
**Pull request by dalthviz**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Base logic to install Kite within Spyder

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: dalthviz

### CODE REVIEW

File: spyder/plugins/completion/kite/utils/install.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            self._installer_name = 'Kite.dmg'
+        else:
+            self._download_url = self.LINUX_URL
+            self._installer_name = 'kite_installer.sh'
+
+    def download_installer(self, progress_reporter=None):
+        """Download the installer in a temporary directory."""
+        temp_dir = gettempdir()
+        path = osp.join(temp_dir, self._installer_name)
+        return urlretrieve(self._download_url, path,
```
Comment (goanpeca) [inline spyder/plugins/completion/kite/utils/install.py]: Is this a GUI blocking call @dalthviz ?

File: spyder/plugins/completion/kite/utils/install.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    LINUX_URL = "https://release.kite.com/dls/linux/current"
+    MAC_URL = "https://release.kite.com/dls/mac/current"
+
+    # Process status
+    NO_STATUS = "No status"
+    DOWNLOADING_SCRIPT = "Downloading Kite script installer"
+    DOWNLOADING_INSTALLER = "Downloading Kite installer"
+    INSTALLING = "Installing Kite"
+    FINISHED = "Install finished"
+    ERRORED = "Error"
```
Comment (goanpeca) [inline spyder/plugins/completion/kite/utils/install.py]: @dalthviz remember to add the translation func.

`NO_STATUS = "No status"` -> ` NO_STATUS = _("No status")`

File: spyder/plugins/completion/kite/utils/status.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+def check_if_kite_installed():
+    """Detect if kite is installed and return the installation path."""
+    path = ''
+    if os.name == 'nt':
+        path = 'C:\\Program Files\\Kite\\kited.exe'
+    elif sys.platform.startswith('linux'):
+        path = osp.expanduser('~/.local/share/kite/kited')
+    elif sys.platform == 'darwin':
+        path = locate_kite_darwin()
+    return osp.exists(osp.realpath(path)), path
```
Comment (goanpeca) [inline spyder/plugins/completion/kite/utils/status.py]: How could we handle custom installations on specific locations?

Comment (dalthviz) [reply spyder/plugins/completion/kite/utils/status.py]: I'm unsure about this

**Comment by dalthviz** [comment]
@ccordoba12 I think this is ready

**Pull request merged**